llvm/lib/FileCheck/FileCheck.cpp

```diff
@@ -1957,11 +1960,12 @@ bool FileCheck::readCheckFile(
   }
 
   // Add an EOF pattern for any trailing --implicit-check-not/CHECK-DAG/-NOTs,
-  // and use the first prefix as a filler for the error message.
+  // and use the prefix from the last/trailing CHECK-NOT for the error message
   if (!DagNotMatches.empty()) {
     CheckStrings->emplace_back(
         Pattern(Check::CheckEOF, PatternContext.get(), LineNumber + 1),
-        *Req.CheckPrefixes.begin(), SMLoc::getFromPointer(Buffer.data()));
+        DagNotMatches.back().DagNotPrefix,
+        SMLoc::getFromPointer(Buffer.data()));
```

Comment:
I think this can remain unchanged now, we won't be using that prefix for the error message once you change printMatch and printNotMatch to handle CheckNot

Reply:
Okay!